inc/az_ulib_ustream_forward_base.h

```diff
@@ -0,0 +1,374 @@
+}
+
+/**
+ * @brief   Returns the remaining size of the ustream_forward.
+ *
+ *  This API returns the number of bytes between the current position and the end of the ustream_forward.
+ *
+ *  The `az_ulib_ustream_forward_get_remaining_size` API shall follow the following minimum requirements:
+ *      - The `get_remaining_size` shall return the number of bytes between the current position
+ *          and the end of the ustream_forward.
+ *      - If the provided interface is `NULL`, the `get_remaining_size` shall fail with
+ *          precondition.
+ *      - If the provided interface is not the implemented ustream_forward type, the `get_remaining_size`
+ *          shall fail with precondition.
+ *      - If the provided size is `NULL`, the `get_remaining_size` shall fail with precondition.
+ *
+ * @param[in]   ustream_forward_instance    The #az_ulib_ustream_forward* with the interface of the ustream_forward.
+ * @param[out]  size                The `size_t* const` to return the remaining number of
+ *                                  `uint8_t` values.
+ *
+ * @pre     \p ustream_forward_instance shall not be `NULL`.
+ * @pre     \p ustream_forward_instance shall be a valid ustream_forward that is the implemented ustream_forward type.
+ * @pre     \p size shall not be `NULL`.
+ *
+ * @return The #az_result with the result of the `get_remaining_size` operation.
+ *      @retval #AZ_OK                        If it succeeded to get the remaining size of the
+ *                                            ustream_forward.
+ *      @retval #AZ_ERROR_ULIB_BUSY           If the resource necessary to get the remaining
+ *                                            size of the ustream_forward is busy.
+ *      @retval #AZ_ERROR_CANCELED            If the `get_remaining_size` was cancelled.
+ *      @retval #AZ_ERROR_NOT_ENOUGH_SPACE    If there is not enough memory to execute the
+ *                                            `get_remaining_size` operation.
+ *      @retval #AZ_ERROR_ULIB_SECURITY       If the `get_remaining_size` was denied for
+ *                                            security reasons.
+ *      @retval #AZ_ERROR_ULIB_SYSTEM         If the `get_remaining_size` operation failed on
+ *                                            the system level.
+ */
+AZ_INLINE az_result
+az_ulib_ustream_forward_get_remaining_size(az_ulib_ustream_forward* ustream_forward_instance, size_t* const size)
+{
+  return ustream_forward_instance->control_block->api->get_remaining_size(ustream_forward_instance, size);
+}
+
+/**
+ * @brief   Release all the resources allocated to control the instance of the ustream_forward.
+ *
+ *  The dispose will release the instance of the ustream_forward and decrement the reference of the
+ *      ustream_forward. If there are no more references to the ustream_forward, the dispose will release all
+ *      resources allocated to control the ustream_forward.
```

Comment:
It is not correct, there is no clone, so, no multiple instances. This API will release the data.

Reply:
good catch. meant to remove this.